pkg/util/tas/tas_assignment.go

```diff
@@ -0,0 +1,259 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tas
+
+import (
+	"iter"
+	"slices"
+
+	"k8s.io/utils/ptr"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta2"
+)
+
+type TopologyAssignment struct {
+	Levels  []string
+	Domains []TopologyDomainAssignment
+}
+
+type TopologyDomainAssignment struct {
+	Values []string
+	Count  int32
+}
+
+func valueAtIndex(values kueue.TopologyAssignmentSliceLevelValues, idx int) string {
+	if univ := values.Universal; univ != nil {
+		return *univ
+	}
+	ind := values.Individual
+	prefix := ptr.Deref(ind.CommonPrefix, "")
+	suffix := ptr.Deref(ind.CommonSuffix, "")
+	return prefix + ind.Roots[idx] + suffix
+}
+
+func countAtIndex(slice kueue.TopologyAssignmentSlice, idx int) int32 {
+	if univ := slice.PodCounts.Universal; univ != nil {
+		return *univ
+	}
+	return slice.PodCounts.Individual[idx]
+}
+
+func ValuesAtLevel(ta *kueue.TopologyAssignment, levelIdx int) iter.Seq[string] {
```

Comment:
It seems this function is always used in the context of nodes which always requires gymnastics around level counting. So it seems it will be more readable to have exported ValuesAtLowestLevel(), and `valuesAtLevel` just internal.